src/helpers/filter-user-segments.js

```diff
@@ -0,0 +1,19 @@
+import _ from "lodash";
+
+const fieldPath = "ship.private_settings.synchronized_segments";
+
+/**
+ * Returns information if the users should be sent in outgoing sync.
+ * This version should filter out all users if the `synchronized_segments`
+ * setting is empty
+ * @param  {Object} ctx The Context Object
+ * @param  {Object} user Hull user object
+ * @return {Boolean}
+ */
+export default function filterUserSegments(ctx, user) {
```

Comment:
@michaloo as discussed earlier let's make the path configurable, and / or allow to pass a list of segment Ids also